rust/src/ffi/base64.rs

```diff
@@ -17,6 +17,7 @@
 
 use std::os::raw::c_uchar;
 use libc::c_ulong;
+use base64::{Engine as _, engine::general_purpose};
```

Comment:
Why do we need `Engine as _` ?

Reply:
Nothing works w/o this. I'm not sure what these are called but I've seen this before.

Reply:
@jasonish do you understand this rust magic ?

Reply:
You should be able to drop the `as _` here.  `Engine` being a trait just needs to be imported, but doesn't need to be bound to a name. This syntax is usually used when binding the trait to a name would conflict with some other name in scope. I don't see a name conflict here tho.